### ISSUE TITLE
await myfunc() === "value" minifies to false (!1)

### ISSUE THREAD

**Issue by Mehuge** [trullock/NUglify#293]
Combination of await and === comparison causes nuglify to substitute with false. Similar happens with !==

Example Code
```javascript
window.XCP = window.XCP || {};
XCP.Global = class {
  async example1() {
    this.myfuncresult = await myfunc() == "value"
  }
  async example2() {
    this.myfuncresult = await myfunc() === "value"
  }
}
```

Resulting Minified code:  The `example2()` method just sets `this.myfuncresult` to false.

```javascript
window.XCP=window.XCP||{};XCP.Global = (class {
async example1(){this.myfuncresult=await myfunc()=="value"}
async example2(){this.myfuncresult=!1}})
})
```

Version: 1.17.3

Code used to run the minifier:
 
```c#
var jsSettings = new CodeSettings();
jsSettings.MinifyCode = true;
jsSettings.LocalRenaming = LocalRenaming.CrunchAll;
var result = Uglify.Js(File.ReadAllText(serverBundle), jsSettings);
File.WriteAllText(serverBundle, result.Code);
```

**Comment by trullock** [comment]
Thanks for the comprehensive ticket, I'll take a look

**Comment by trullock** [comment]
This repro can be simplified to 

```
async function myfunc()
{
	return await myfunc() === "value";
}
```

The issues is how `await` is evaluated wrt `===`, it thinks the return type is always a number, so they are never strictly equal in this case, hence getting `!1`

**Comment by trullock** [comment]
Fixed in 1.17.7
